src/testing/vortex/workload.zig

```diff
@@ -0,0 +1,432 @@
+//! This workload runs in an loop, generating and executing operations on a cluster through a
+//! _driver_.
+//!
+//! Any sucessful operations are reconciled with a model, tracking what accounts exist. Future
+//! operations are generated based on this model.
+//!
+//! After every operation, all accounts are queried, and basic invariants are checked.
+//!
+//! The workload and drivers communicate with a binary protocol over stdio. The protocol is based
+//! on the extern structs in `src/tigerbeetle.zig` and `src/state_machine.zig`, and it works like
+//! this:
+//!
+//! 1. Workload sends a request, which is:
+//!    * the _operation_ (1 byte),
+//!    * the _event count_ (4 bytes), and
+//!    * the events (event count * size of event).
+//! 2. The driver uses its client to submit those events. When receiving results, it sends them
+//!    back on its stdout as:
+//!    * the _operation_ (1 byte)
+//!    * the _result count_ (4 bytes), and
+//!    * the results (resulgt count * size of result pair), where each pair holds an index and a
+//!      result enum value (see `src/tigerbeetle.zig`)
+//! 3. The workload receives the results, and expects them to be of the same operation type as
+//!    originally requested. There might be fewer results than events, because clients can omit
+//!    .ok results.
+
+const std = @import("std");
+const arbitrary = @import("./arbitrary.zig");
+const tb = @import("../../tigerbeetle.zig");
+const constants = @import("../../constants.zig");
+const StateMachineType = @import("../../state_machine.zig").StateMachineType;
+
+const log = std.log.scoped(.workload);
+const assert = std.debug.assert;
+const testing = std.testing;
+const StateMachine = StateMachineType(void, constants.state_machine_config);
+
+const events_count_max = 8190;
+const pending_transfers_count_max = 1024;
+const accounts_count_max = 128;
+
+const DriverStdio = struct { input: std.fs.File, output: std.fs.File };
+
+pub fn main(
+    allocator: std.mem.Allocator,
+    driver: *const DriverStdio,
+) !void {
+    var model = Model{
```

Comment:
It's OK to use dynamic allocation here for now, but I'd seriously consider refactoring
this to static alloc, to make code _easier_ to understand. Static alloc is harder to code for
the first time, as you need to actually _understand_ how many accounts/pending transfers you
could have, but, once understood and encoded in the source code, this understanding can be
re-used for free by future readers.

In terms of code, that's s/AutoHashMap/AutoHashMapUnmanaged